src/ISA/ISA/ArcTypes/ArcAssay.fs

```diff
@@ -24,26 +82,349 @@ type ArcAssay =
         (technologyType : OntologyAnnotation option)
         (technologyPlatform : string option)
-        (sheets : ArcTable list option)
+        (tables : ResizeArray<ArcTable>)
         (performers : Person list option)
         (comments : Comment list option) = 
         {
-                ID = id
-                FileName = fleName
-                MeasurementType = measurementType
-                TechnologyType = technologyType
-                TechnologyPlatform = technologyPlatform
-                Sheets = sheets
-                Performers = performers
-                Comments = comments
-            }
-      
+            ID = id
+            FileName = fleName
+            MeasurementType = measurementType
+            TechnologyType = technologyType
+            TechnologyPlatform = technologyPlatform
+            Tables = tables
+            Performers = performers
+            Comments = comments
+        }
+
+    member this.TableCount 
+        with get() = this.Tables.Count
 
+    member this.TableNames 
+        with get() = 
+            [for s in this.Tables do yield s.Name]
 
     [<NamedParams>]
-    static member create (?ID : URI, ?FileName : string, ?MeasurementType : OntologyAnnotation, ?TechnologyType : OntologyAnnotation, ?TechnologyPlatform : string, ?Sheets : ArcTable list, ?Performers : Person list, ?Comments : Comment list) = 
+    static member create (?ID : URI, ?FileName : string, ?MeasurementType : OntologyAnnotation, ?TechnologyType : OntologyAnnotation, ?TechnologyPlatform : string, ?Sheets : ResizeArray<ArcTable>, ?Performers : Person list, ?Comments : Comment list) = 
+        let Sheets = ResizeArray()
         ArcAssay.make ID FileName MeasurementType TechnologyType TechnologyPlatform Sheets Performers Comments
 
+
+    // - Table API - //
+    // remark should this return ArcTable?
+    member this.AddTable(?table:ArcTable, ?index: int) = 
+        let index = defaultArg index this.TableCount
+        let table =
+            let createName() = getNextAutoGeneratedTableName this.TableNames
+            defaultArg table (createName())
+        SanityChecks.validateSheetIndex index true this.Tables
+        SanityChecks.validateNewNameUnique table.Name this.TableNames
+        this.Tables.Insert(index, table)
+
+    member this.AddTables(tables:seq<ArcTable>, ?index: int) = 
+        let index = defaultArg index this.TableCount
+        SanityChecks.validateSheetIndex index true this.Tables
+        SanityChecks.validateNewNamesUnique (tables |> Seq.map (fun x -> x.Name)) this.TableNames
+        this.Tables.InsertRange(index, tables)
+
+    // - Table API - //
+    member this.GetTableAt(index:int) : ArcTable =
+        SanityChecks.validateSheetIndex index false this.Tables
+        this.Tables.[index]
+
+    static member getTableAt(index:int) : ArcAssay -> ArcTable =
+        fun (assay:ArcAssay) ->
+            let newAssay = assay.Copy()
+            newAssay.GetTableAt(index)
```

Comment:
- add comment to make clear that the user receives a copied object
- copy table, not assay